protos/flyteidl/datacatalog/datacatalog.proto

```diff
@@ -150,6 +158,38 @@ message ListDatasetsResponse {
     string next_token = 2;
 }
 
+// Get the Artifact or try to reserve a spot if the Artifact does not exist.
+message GetOrReserveArtifactRequest {
+    DatasetID dataset_id = 1;
+    string tag_name = 2;
+
+    // How long you can occupy the spot.
```

Comment:
Another interesting question over here is, what happens if its been a long time and the process 2 now has been granted a reservation. Lets assume process 3 comes along, will it get an Already_in_progress? If so, we will have to store both the reservations.

I think it is worth documenting these semantics clearly.

Also  this should be reworded
```suggestion
    // Minimum duration to wait from the time when the original reservation was made to consider that the original reservation has been timed-out.
```